pkg/actions/cluster/update.go

```diff
@@ -0,0 +1,97 @@
+)
+
+func updateVersion(dryRun bool, cfg *api.ClusterConfig, currentVersion string, ctl *eks.ClusterProvider) error {
+	msgNodeGroupsAndAddons := "you will need to follow the upgrade procedure for all of nodegroups and add-ons"
+	cmdutils.LogIntendedAction(dryRun, "upgrade cluster %q control plane from current version %q to %q", cfg.Metadata.Name, currentVersion, cfg.Metadata.Version)
+	if !dryRun {
+		if err := ctl.UpdateClusterVersionBlocking(cfg); err != nil {
+			return err
+		}
+		logger.Success("cluster %q control plane has been upgraded to version %q", cfg.Metadata.Name, cfg.Metadata.Version)
+		logger.Info(msgNodeGroupsAndAddons)
+	}
+	return nil
+}
+
+func requiresVersionUpgrade(clusterMeta *api.ClusterMeta, currentEKSVersion string) (bool, error) {
+	nextVersion, err := getNextVersion(currentEKSVersion)
+	if err != nil {
+		return false, err
+	}
+
+	// If the version was not specified default to the next Kubernetes version and assume the user intended to upgrade if possible
+	// also support "auto" as version (see #2461)
+	if clusterMeta.Version == "" || clusterMeta.Version == "auto" {
+		if api.IsSupportedVersion(nextVersion) {
+			clusterMeta.Version = nextVersion
+			return true, nil
+		}
+
+		// There is no new version, stay in the current one
+		clusterMeta.Version = currentEKSVersion
+		return false, nil
+	}
+
+	if c, err := utils.CompareVersions(clusterMeta.Version, currentEKSVersion); err != nil {
+		return false, errors.Wrap(err, "couldn't compare versions for upgrade")
+	} else if c < 0 {
+		return false, fmt.Errorf("cannot upgrade to a lower version. Found given target version %q, current cluster version %q", clusterMeta.Version, currentEKSVersion)
+	}
+
+	if api.IsDeprecatedVersion(clusterMeta.Version) {
+		return false, fmt.Errorf("control plane version %q has been deprecated", clusterMeta.Version)
+	}
+
+	if !api.IsSupportedVersion(clusterMeta.Version) {
+		return false, fmt.Errorf("control plane version %q is not known to this version of eksctl, try to upgrade eksctl first", clusterMeta.Version)
+	}
+
+	if clusterMeta.Version == currentEKSVersion {
+		return false, nil
+	}
+
+	if clusterMeta.Version == nextVersion {
+		return true, nil
+	}
+
+	return false, fmt.Errorf(
+		"upgrading more than one version at a time is not supported. Found upgrade from %q to %q. Please upgrade to %q first",
+		currentEKSVersion,
+		clusterMeta.Version,
+		nextVersion)
+}
+
+func getNextVersion(currentVersion string) (string, error) {
+	switch currentVersion {
+	case "":
+		return "", errors.New("unable to get control plane version")
+	case api.Version1_12:
+		return api.Version1_13, nil
+	case api.Version1_13:
+		return api.Version1_14, nil
+	case api.Version1_14:
+		return api.Version1_15, nil
+	case api.Version1_15:
+		return api.Version1_16, nil
+	case api.Version1_16:
+		return api.Version1_17, nil
+	case api.Version1_17:
+		return api.Version1_18, nil
+	case api.Version1_18:
+		return api.Version1_19, nil
```

Comment:
A couple of questions here:
1. How come you started at 1.12? We list versions as far back as 1.10 are listed as unsupported.
1. What is the UX for people whose next version is unsupported? I can see we give them this error, but do we suggest a next course of action? I can see people thinking "oh i'll just bump n versions then" which we also don't allow.

Reply:
Great question, I'm not sure regarding 2). I know that we do manually add versions to eksctl when new versions of kubernetes are released. 

> I can see people thinking "oh i'll just bump n versions then" which we also don't allow.

Do you mean kubernetes versions? or CLI versions?

Reply:
kubernetes

Reply:
I'm assuming the EKS API will reject it, although I haven't tried.

Reply:
hmm i imagine that looks pretty. meh, we can wait until somebody tries it and gets confused